ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/StreamedQueryResource.java

```diff
@@ -396,10 +399,16 @@ private EndpointResponse handleQuery(final KsqlSecurityContext securityContext,
           );
         }
         case KSTREAM: {
-          throw new KsqlStatementException(
-              "Pull queries are not supported on streams."
-                  + PullQueryValidator.PULL_QUERY_SYNTAX_HELP,
-              statement.getStatementText()
+          final SessionConfig sessionConfig = SessionConfig.of(ksqlConfig, configProperties);
```

Comment:
Is there any reason not to allow this for tables which are not materialized? I guess there's the risk that they may not know the difference and assume this will be as performant as a state store lookup, but it seems like this should more or less work right out of the box for those as well.

Reply:
That's an interesting idea. I think this would have to be a separate feature, though, since the semantics are different. For non-materialized tables, we'd have to read the whole topic (like we're doing here), but collect it in a data structure instead of emitting results as we go, and then compute the query and emit the results at the end.

Reply:
Hmm, that's a good point.  If we outputted a series of "A": 1 and then "A":2 as it changed, that would violate the semantics of what we would expect from a state store lookup. I agree that would take more work so that can be another feature.  That said, it is slightly confusing what's queryable with pull queries and what's not.

Reply:
Yep, it's a good point. The feature request for stream pull queries mentioned "filling out the missing quadrant", but what you're pointing out is that there's still a missing quadrant: pull queries against non-materialized tables.